crates/consensus/common/src/validation.rs

```diff
@@ -848,4 +848,78 @@ mod tests {
             }))
         );
     }
+
+    #[test]
+    fn test_valid_gas_limit_increase() {
+        let parent = SealedHeader {
+            header: Header { gas_limit: 1024 * 10, ..Default::default() },
+            ..Default::default()
+        };
+        let child = SealedHeader {
+            header: Header { gas_limit: parent.header.gas_limit + 5, ..Default::default() },
+            ..Default::default()
+        };
+        let chain_spec = ChainSpec::default();
+
+        assert_eq!(check_gas_limit(&parent, &child, &chain_spec), Ok(()));
+    }
+
+    #[test]
+    fn test_london_hardfork_adjustment() {
+        let parent = SealedHeader {
+            header: Header { gas_limit: 1024 * 10, ..Default::default() },
+            ..Default::default()
+        };
+        let child = SealedHeader {
+            header: Header { gas_limit: parent.header.gas_limit, ..Default::default() },
+            ..Default::default()
+        };
+        let mut chain_spec = ChainSpec::default();
+        chain_spec.hardforks.insert(Hardfork::London, ForkCondition::Block(1));
+
+        assert_eq!(check_gas_limit(&parent, &child, &chain_spec), Ok(()));
+    }
+
+    fn create_header(gas_limit: u64) -> SealedHeader {
+        SealedHeader { header: Header { gas_limit, ..Default::default() }, ..Default::default() }
+    }
+
+    #[test]
+    fn test_invalid_gas_limit_increase_exceeding_limit() {
+        let parent = create_header(1024 * 10);
```

Comment:
not sure we need a method just for this, we can probably create the headers directly with the specified gas limits, like with the first few tests

Reply:
I just added it for better readability and we can us it if we want to use SealedHeader with default header it become more easy

Reply:
we're still using `create_header` here, would recommend removing the method and creating the headers manually in tests